gegelatilib/include/learn/learningParameters.h

```diff
@@ -80,6 +84,18 @@ namespace Learn {
 		*/
 		size_t nbThreads = std::thread::hardware_concurrency();
 	} LearningParameters;
+
+    /// Reads a given json file and puts the derivative tree in root
+    void readConfigFile(const char* path, Json::Value &root);
+
+    /// Puts the parameters described in the derivative tree root into params
+    void setAllParamsFrom(const Json::Value& root, LearningParameters *params);
+
+    /// Loads a given json file and puts the parameters it contains in params
+    void loadParametersFromJson(const char* path, LearningParameters *params);
+
+    /// Given a parameter name, sets its value in params
+    void setParameterFromString(LearningParameters * params, std::string& param, double value);
```

Comment:
Move JSON reading code to separate file.
The LearningParameters class may exist withouth its parser. Hence, I'd move the parsing code outside of this file, to a dedicated place.